src/coprocessor/dag/batch/executors/index_scan_executor.rs

```
@@ -153,6 +155,7 @@ impl super::util::scan_executor::ScanExecutorImpl for IndexScanExecutorImpl {
             ));
         }
 
+        assert_eq!(columns.len(), columns_len);
```

I think panic is bad, maybe return an error will be better.

---

Does it seem it would never panic unless we do not process correctly when building the `BatchIndexScan`? So could we remove it?

---

> Seems it would never panic unless we do not process correctly when building the BatchIndexScan? So could we remove it?

So this is an assertion.